### PR TITLE
push: Fix docker-compose.dev.yml serialization ("should be object,null" error)

### PR DESCRIPTION
**Pull request by pdcastro**
Resolves: #2226 
Change-type: patch

The following `volumes` section is typical in a `docker-compose.yml` file of a balena application:
```
volumes:
  resin-data:
```

Before this PR, when `docker-compose.dev.yml` was merged with `docker-compose.yml` and then serialized, the above would become:

```
volumes:
  resin-data: null
```

And then `balena push` would result in:
```
$ balena push 192.168.1.46
[Info]    Starting build on device 192.168.1.46
[Info]    Docker compose dev overlay detected (docker-compose.dev.yml) - merging.
data/volumes/resin-data should be object,null
```

According to the Yaml spec, `resin-data:` means the same as `resin-data: null`, so that error was unexpected. I thought it might thus be a case of incorrect validation (a case of fixing the `resin-compose-parse` module) but, on investigation, the schema of `docker-compose.yml` already defines `volume` as `"type": ["object", "null"]`, which I believe is correct:

https://github.com/balena-io-modules/resin-compose-parse/blob/v2.1.3/src/schemas/config_schema_v2.1.json#L34
```
    "volume": {
      "id": "#/definitions/volume",
      "type": ["object", "null"],
```

So I don't know if there is something to fix there. In any case, I found that `js-yaml` v4.0.0 introduces a new serialization option `styles: { '!!null': 'empty' }` that causes the serialization to produce:
```
volumes:
  resin-data:
```

Which then avoids the error of issue #2226. So that's what I did in this PR.
cc: @srlowe

### CODE REVIEW

File: automation/update-module.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ const getUpstreams = async () => {
 	const repoYaml = fs.readFileSync(__dirname + '/../repo.yml', 'utf8');
 
 	const yaml = await import('js-yaml');
-	const { upstream } = yaml.safeLoad(repoYaml) as {
+	const { upstream } = yaml.load(repoYaml) as {
```
Comment (pdcastro) [inline automation/update-module.ts:61]: `safeLoad` was removed in `js-yaml` v4.0.0 (breaking change), which is updated by this PR.

File: lib/utils/compose_ts.ts [hunk 2]
```diff
@@ -179,13 +179,18 @@ async function mergeDevComposeOverlay(
...
 		try {
-			const yaml = await import('js-yaml');
-			const compose = yaml.load(composeStr);
-			const devOverlay = yaml.load(await fs.readFile(devOverlayPath, 'utf8'));
+			const compose = loadObj(composeStr);
+			const devOverlay = loadObj(await fs.readFile(devOverlayPath, 'utf8'));
 			// We only want to merge the services section
 			compose.services = { ...compose.services, ...devOverlay.services };
-			composeStr = yaml.dump(compose);
+			composeStr = yaml.dump(compose, { styles: { '!!null': 'empty' } });
```
Comment (pdcastro) [inline lib/utils/compose_ts.ts:193]: `{ styles: { '!!null': 'empty' } }` is what fixes issue #2226. This option was introduced in `js-yaml` v4.0.0, which is why this PR updates `js-yaml`.

**Comment by pdcastro** [comment]
@balena-ci I self-certify!

**Comment by pdcastro** [comment]
@srlowe, do you think that the root cause of #2226 is in `resin-compose-parse`? I was going to add a test case to this PR to ensure `resin-data:` instead of `resin-data: null`, but then I thought that perhaps we should make `resin-data: null` work, instead of avoiding it.
